Error on startup with command: 'start.sh dev'

▶ docker logs hbp_development_hasura-backend-plus_run_7ab302bc0214
yarn run v1.22.4
$ ts-node-dev -r tsconfig-paths/register --no-deps --respawn --poll --interval 1000 --no-notify src/start.ts
Using ts-node version 8.8.2, typescript version 3.8.3
Error: Cannot find module '/app/dist/shared/config'
Require stack:
- /app/src/start.ts
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1029:15)
    at Function.Module._resolveFilename (/app/node_modules/tsconfig-paths/lib/register.js:75:40)
    at Function.Module._resolveFilename (/app/node_modules/module-alias/index.js:49:29)
    at Function.Module._load (internal/modules/cjs/loader.js:898:27)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
    at require (internal/modules/cjs/helpers.js:73:18)
    at Object.<anonymous> (/app/src/start.ts:2:1)
    at Module._compile (internal/modules/cjs/loader.js:1200:30)
    at Module._compile (/app/node_modules/source-map-support/source-map-support.js:541:25)
    at Module.m._compile (/tmp/ts-node-dev-hook-9832059318274387.js:57:25)
[ERROR] 21:36:23 Error: Cannot find module '/app/dist/shared/config'
Require stack:
- /app/src/start.ts

--- a/Dockerfile
+++ b/Dockerfile
@@ -11,7 +11,7 @@ RUN yarn install
 
 COPY . .
 
-RUN ./node_modules/typescript/bin/tsc -p .
+RUN yarn build
 
 HEALTHCHECK --interval=60s --timeout=2s --retries=3 CMD wget ${HOST}:${PORT}/healthz -q -O - > /dev/null 2>&1
 

--- a/package.json
+++ b/package.json
@@ -8,8 +8,9 @@
   "version": "2.0.0",
   "main": "src/start.ts",
   "scripts": {
-    "dev": "ts-node-dev -r tsconfig-paths/register --no-notify src/start.ts",
-    "dev:docker": "ts-node-dev -r tsconfig-paths/register --no-deps --respawn --poll --interval 1000 --no-notify src/start.ts",
+    "build": "tsc",
+    "dev": "ts-node-dev -r tsconfig-paths/register --no-notify src/ts-start.ts",
+    "dev:docker": "ts-node-dev -r tsconfig-paths/register --no-deps --respawn --poll --interval 1000 --no-notify src/ts-start.ts",
     "docs:build": "vuepress build docs",
     "docs:dev": "vuepress dev docs",
     "start": "node -r ./prod-paths.js ./dist/start.js",

--- a/src/migrate.ts
+++ b/src/migrate.ts
@@ -30,16 +30,21 @@ const waitFor = async (path: string, attempts = 240): Promise<void> => {
 }
 
 const hasuraConsole = async (action: string): Promise<void> => {
-  const child = spawn('./node_modules/.bin/hasura', [
-    ...action.split(' '),
-    '--log-level',
-    LOG_LEVEL,
-    '--skip-update-check',
-    '--project',
-    TEMP_MIGRATION_DIR
-  ])
-  for await (const data of child.stdout) {
-    process.stdout.write(data.toString())
+  try {
+    const child = spawn('./node_modules/.bin/hasura', [
+      ...action.split(' '),
+      '--log-level',
+      LOG_LEVEL,
+      '--skip-update-check',
+      '--project',
+      TEMP_MIGRATION_DIR
+    ])
+    for await (const data of child.stdout) {
+      process.stdout.write(data.toString())
+    }
+  } catch (error) {
+    console.log('Error in starting hasura cli')
+    console.log(error)
   }
 }
 

--- a/src/start.ts
+++ b/src/start.ts
@@ -1,23 +1,2 @@
 require('module-alias/register')
-import { HOST, PORT, AUTO_MIGRATE } from '@shared/config'
-import { app } from './server'
-import migrate from './migrate'
-
-const start = async (): Promise<void> => {
-  if (AUTO_MIGRATE) {
-    const migrationSetup = {
-      migrations: AUTO_MIGRATE === 'v1' ? './migrations-v1' : './migrations'
-      // metadata: './metadata'
-    }
-    await migrate(migrationSetup)
-  }
-  app.listen(PORT, HOST, () => {
-    if (HOST) {
-      console.log(`Running on http://${HOST}:${PORT}`)
-    } else {
-      console.log(`Running on port ${PORT}`)
-    }
-  })
-}
-
-start()
+import './ts-start'

--- a/src/ts-start.ts
+++ b/src/ts-start.ts
@@ -1,0 +1,22 @@
+import { HOST, PORT, AUTO_MIGRATE } from '@shared/config'
+import { app } from './server'
+import migrate from './migrate'
+
+const start = async (): Promise<void> => {
+  if (AUTO_MIGRATE) {
+    const migrationSetup = {
+      migrations: AUTO_MIGRATE === 'v1' ? './migrations-v1' : './migrations'
+      // metadata: './metadata'
+    }
+    await migrate(migrationSetup)
+  }
+  app.listen(PORT, HOST, () => {
+    if (HOST) {
+      console.log(`Running on http://${HOST}:${PORT}`)
+    } else {
+      console.log(`Running on port ${PORT}`)
+    }
+  })
+}
+
+start()

